src/celeritas/field/FieldPropagator.hh

```diff
@@ -162,7 +162,11 @@ CELER_FUNCTION auto FieldPropagator<DriverT>::operator()(real_type step)
         // Do a detailed check boundary check from the start position toward
         // the substep end point. Travel to the end of the chord, plus a little
         // extra.
-        geo_.set_dir(chord.dir);
+        if (chord.length >= driver_.minimum_step())
```

Comment:
Logically this is somewhat redundant as always `remaining > driver_.minimum_step()` and the chord.length should be nontrivial even though it may be smaller than driver_.minimum_step() unless the field gradient is very big.  Not sure why this extra protection helps (i.e, what is wrong if the final direction is very close to the original direction?)

Reply:
Since `chord.length` is always less than or equal to `remaining`, it can *sometimes* be less than the driver minimum step (e.g. in a strong magnetic field/low particle energy where the remaining step is just *barely* greater than the driver minimum step).

Additionally, on the first run through the loop (as *always* happens with the failing cases) even the `remaining`  can be less than minimum step.